### PR TITLE
Add conversion from `Path` for `EntityTypeQueryPath` and `LinkTypeQueryPath`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Similar to the conversion to `DataTypeQueryPath` and `PropertyTypeQueryPath`, this adds the same for `EntityTypeQueryPath` and `LinkTypeQueryPath`

## 🔗 Related links

- [Asana task: entity types](https://app.asana.com/0/1203007126736607/1203167266370356/f) _(internal)_
- [Asana task: link types](https://app.asana.com/0/1203007126736607/1203167266370357/f) _(internal)_

## 🚫 Blocked by

- #1221 
- #1222 

## 🔍 What does this change?

Similar to #1204 and #1221, this adds a similar conversion for `EntityTypeQueryPath` and `LinkTypeQueryPath` and some tests

## 🐾 Next steps

Use the ontology queries in REST endpoints: [Asana task](https://app.asana.com/0/0/1203205825668099/f) (_internal_)

## 🛡 What tests cover this?

Similar to #1204 and #1221 tests were added to test the conversion

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/ontology/entity_type.rs [hunk 1]
```diff
@@ -30,7 +38,224 @@ impl QueryRecord for EntityType {
...
+    Description,
+    Default,
+    Examples,
+    Properties,
+    Required,
+    Links,
+    RequiredLinks,
+}
+
+/// Deserializes a [`EntityTypeQueryPath`] from a string sequence.
```
Comment (Alfred-Mountfield) [inline packages/graph/hash_graph/lib/graph/src/ontology/entity_type.rs:67]:
```suggestion
/// Deserializes an [`EntityTypeQueryPath`] from a string sequence.
```

**Review by Alfred-Mountfield — COMMENTED** [review]
Tiny typo, will commit directly

**Review by Alfred-Mountfield — APPROVED** [review]
Looks good to me, thanks! 

Bit worried about the sheer amount of duplication we're seeing with this code, but sometimes that can't be avoided. Maybe we can look into macros if this becomes hard to maintain

**Comment by TimDiekmann** [comment]
Thanks Alfie!
A big chunk of duplication will be removed with #1259 🙂

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**